src/Microsoft.Extensions.Hosting/HostBuilder.cs

```diff
@@ -150,6 +154,69 @@ private string ResolveContentRootPath(string contentRootPath, string basePath)
             return Path.Combine(Path.GetFullPath(basePath), contentRootPath);
         }
 
+        private void CreateHostingApplicationData()
+        {
+            var appDataPath = ResolveApplicationDataPath(_hostConfiguration[HostDefaults.ApplicationDataKey]);
+            _hostingApplicationData = new HostingApplicationData();
+            if (appDataPath != null)
+            {
+                _hostingApplicationData.ApplicationDataPath = appDataPath;
+                _hostingApplicationData.ApplicationDataFileProvider = new PhysicalFileProvider(appDataPath);
+            }
+        }
+
+        private string ResolveApplicationDataPath(string applicationDataPath)
+        {
+            var directoryInfo = GetOrCreateDirectory(applicationDataPath);
+            if (directoryInfo != null)
+            {
+                return directoryInfo.FullName;
+            }
+
+            // Environment.GetFolderPath returns null if the user profile isn't loaded.
+            var localAppDataFromSystemPath = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            var localAppDataFromEnvPath = Environment.GetEnvironmentVariable("LOCALAPPDATA");
+            var userProfilePath = Environment.GetEnvironmentVariable("USERPROFILE");
+            var homePath = Environment.GetEnvironmentVariable("HOME");
+
+            // To preserve backwards-compatibility with 1.x, Environment.SpecialFolder.LocalApplicationData
+            // cannot take precedence over $LOCALAPPDATA and $HOME/.aspnet on non-Windows platforms.
+            // We do this in here too to keep consistency with the locations where data protection stores keys.
+            directoryInfo = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+                GetOrCreateDirectory(GetFolderPath(localAppDataFromSystemPath, "ASP.NET")) :
+                null;
+
+            directoryInfo = directoryInfo ??
```

Comment:
Show this search list to @DamianEdwards and explain to him what kind of files you'd expect to store here.